src/bridge/bridge.c

```diff
@@ -173,66 +173,41 @@ send_init_command (CockpitTransport *transport,
   g_bytes_unref (bytes);
 }
 
-static void
-setup_dbus_daemon (gpointer addrfd)
-{
-  g_unsetenv ("G_DEBUG");
-  cockpit_unix_fd_close_all (3, GPOINTER_TO_INT (addrfd));
-}
-
-static GPid
+static GSubprocess *
 start_dbus_daemon (void)
 {
-  GError *error = NULL;
-  GString *address = NULL;
+  g_autoptr(GError) error = NULL;
   gchar *line;
   gsize len;
   gssize ret;
-  GPid pid = 0;
-  gchar *print_address = NULL;
+  GSubprocess *process = NULL;
```

Comment:
if you're gonna be juggling this all around anyway, can you move it to the style of declaration the variable at its site of first use?

Reply:
Done.